connectors/bitcoin.go

```diff
@@ -4,29 +4,155 @@ import (
 	"bytes"
 	"encoding/binary"
 	"fmt"
+	"time"
 
+	"github.com/btcsuite/btcd/chaincfg"
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/rpcclient"
 	"github.com/btcsuite/btcd/wire"
 	"github.com/btcsuite/btcutil"
 	"github.com/btcsuite/btcutil/bloom"
+	log "github.com/sirupsen/logrus"
 )
 
-// Computes the serialized partial merkle tree of a transaction in a block
-// the format of the serialized PMT is:
+type BTC struct {
+	c      *rpcclient.Client
+	chans  map[string]*chan bool
+	params chaincfg.Params
+}
+
+type AddressWatcher interface {
+	OnNewConfirmation(txHash string, confirmations int64, amount float64)
+}
+
+func NewBTC() *BTC {
+	return &BTC{chans: make(map[string]*chan bool)}
+}
+
+func (btc *BTC) Connect(endpoint string, username string, password string, network string) error {
+	switch network {
+	case "mainnet":
+		btc.params = chaincfg.MainNetParams
+	case "testnet":
+		btc.params = chaincfg.TestNet3Params
+	default:
+		return fmt.Errorf("invalid network name: %v", network)
+	}
+	config := rpcclient.ConnConfig{
+		Host:         endpoint,
+		User:         username,
+		Pass:         password,
+		Params:       btc.params.Name,
+		DisableTLS:   true,
+		HTTPPostMode: true,
+	}
+	c, err := rpcclient.New(&config, nil)
+	if err != nil {
+		return fmt.Errorf("RPC client error: %v", err)
+	}
+	btc.c = c
+	return nil
+}
+
+func (btc *BTC) AddAddressWatcher(address string, interval time.Duration, w AddressWatcher) error {
+	btcAddr, err := btcutil.DecodeAddress(address, &btc.params)
+	if err != nil {
+		return fmt.Errorf("error decoding address %v: %v", address, err)
+	}
+
+	err = btc.c.ImportAddressRescan(address, "", false)
+	if err != nil {
+		return fmt.Errorf("error importing address %v: %v", address, err)
```

Comment:
shouldn't the original error message be enough here?

Reply:
I am not sure, because sometimes the error returned by the call does not give context on what exactly failed. For example, it could say "method not found", and then you wouldn't know which is the method that is not found.